src/cpuid/src/cpuid_ffi.rs

```diff
@@ -13,6 +13,16 @@ use serde::{Deserialize, Serialize};
 
 use super::{CpuidEntry, CpuidKey};
 
+/// converts `u32` to `usize`.
+fn from_u32(x: u32) -> usize {
+    #[cfg(not(any(target_pointer_width = "32", target_pointer_width = "64")))]
+    std::compile_error!("Firecracker does not support < 32-bit hosts.");
+
+    // JUSTIFICATION: Firecracker does not support < 32-bit hosts.
+    // SAFETY: The preceding `cfg` guarantees this is safe.
+    unsafe { usize::try_from(x).unwrap_unchecked() }
+}
+
```

Comment:
> there should be no performance impact, as the TryFrom implementation always returns Ok(x as usize) on 32 and 64 bit platforms (by means of conditional compilation).

`unwrap` is not const so there is work to be done at runtime here.

The internal `cfg`s can change as this is not public, it is not guaranteed. 

The remaining reasons are the same as https://rust-lang.github.io/rust-clippy/master/#unwrap_used:

> It is better to handle the None or Err case, or at least call .expect(_) with a more helpful message. Still, for a lot of quick-and-dirty code, unwrap is a good choice, which is why this lint is Allow by default.
> 
> result.unwrap() will let the thread panic on Err values. Normally, you want to implement more sophisticated error handling, and propagate errors upwards with ? operator.
> 
> Even if you want to panic on errors, not all Errors implement good messages on display. Therefore, it may be beneficial to look at the places where they may get displayed. Activate this lint to do just that.

Reply:
> unwrap is not const so there is work to be done at runtime here.

`from_u32` is not called from any const-evaluated context (and is not const itself), so it will get executed at runtime anway. 

> The internal cfgs can change as this is not public, it is not guaranteed.

If the `impl TryFrom<u32> for usize` implementation on a 32 or 64 bit platform every changes to potentially return `Err`, then the code as it is written right now would trigger undefined behavior. 

 >> It is better to handle the None or Err case, or at least call .expect(_) with a more helpful message. Still, for a lot of quick-and-dirty code, unwrap is a good choice, which is why this lint is Allow by default.
 >> result.unwrap() will let the thread panic on Err values. Normally, you want to implement more sophisticated error handling, and propagate errors upwards with ? operator.
>> Even if you want to panic on errors, not all Errors implement good messages on display. Therefore, it may be beneficial to look at the places where they may get displayed. Activate this lint to do just that.

This just argues in favor of using more sophisticated error handling, which is orthogonal. To me it would make more sense to adjust the `cfg` on the cpuid module to be `x86 AND pointer width large enough` (which would have the same compilation error as this) and use .unwrap here.